daemon-tests/src/flow.rs

```diff
@@ -14,17 +13,19 @@ const NEXT_WAIT_TIME: Duration = Duration::from_secs(if cfg!(debug_assertions) {
 
 /// Wait and return next non-empty maker offers with `contract_symbol`
 pub async fn next_maker_offers(
-    rx_a: &mut OffersFeed,
-    rx_b: &mut OffersFeed,
+    rx_a: &mut watch::Receiver<MakerOffers>,
+    rx_b: &mut watch::Receiver<MakerOffers>,
     contract_symbol: &ContractSymbol,
 ) -> Result<(MakerOffers, MakerOffers)> {
-    let (mut rx_a, mut rx_b) = match contract_symbol {
-        ContractSymbol::BtcUsd => (rx_a.btc_usd.clone(), rx_b.btc_usd.clone()),
-        ContractSymbol::EthUsd => (rx_a.eth_usd.clone(), rx_b.eth_usd.clone()),
-    };
+    let mut rx_a = rx_a.clone();
+    let mut rx_b = rx_b.clone();
 
     let non_empty_offer = |offer: MakerOffers| {
-        if offer.long.is_some() && offer.short.is_some() {
+        if offer.long.is_some()
```

Comment:
🛠️ I think we should avoid `unwrap`s in the codebase .

Maybe try n `match` for the `Some` case with an additional `if`.
Or maybe an `if let Some(offer) = offer.long` 

---

Is it intentional that we always send out long and short together? (I guess yes)

Reply:
Happy to change, but i figured I can safely unwrap here since I've checked before for is_some()

Reply:
fixed in https://github.com/itchysats/itchysats/pull/2679/commits/8089dac7d577f12f3bb2688b9fefc586a4b64e27